### ISSUE TITLE
Request Failed: Unable to connect to Deluge, please check your settings

### ISSUE THREAD

**Issue by rafisher79** [Radarr/Radarr#2502]
Has worked fine for months, but after recent update it stopped. Deluge tests fine in the download client settings, but when selecting a torrent, it says that it can't connect.

Version
0.2.0.935

Mono Version
5.8.0.108 (tarball Fri Jan 19 18:15:21 UTC 2018)

NzbDrone.Core.Download.Clients.DownloadClientException: Unable to connect to Deluge, please check your settings ---> System.Net.WebException: The request timed out
  at System.Net.HttpWebRequest.EndGetResponse (System.IAsyncResult asyncResult) [0x00049] in <a63b52a3a7644f60a95b388cb72441c2>:0 
  at System.Net.HttpWebRequest.GetResponse () [0x0000e] in <a63b52a3a7644f60a95b388cb72441c2>:0 
  at NzbDrone.Common.Http.Dispatchers.ManagedHttpDispatcher.GetResponse (NzbDrone.Common.Http.HttpRequest request, System.Net.CookieContainer cookies) [0x00130] in C:\projects\radarr-usby1\src\NzbDrone.Common\Http\Dispatchers\ManagedHttpDispatcher.cs:74 
  at NzbDrone.Common.Http.Dispatchers.FallbackHttpDispatcher.GetResponse (NzbDrone.Common.Http.HttpRequest request, System.Net.CookieContainer cookies) [0x000cc] in C:\projects\radarr-usby1\src\NzbDrone.Common\Http\Dispatchers\FallbackHttpDispatcher.cs:58 
  at NzbDrone.Common.Http.HttpClient.Execute (NzbDrone.Common.Http.HttpRequest request) [0x0007e] in C:\projects\radarr-usby1\src\NzbDrone.Common\Http\HttpClient.cs:63 
  at NzbDrone.Core.Download.Clients.Deluge.DelugeProxy.ExecuteRequest[TResult] (NzbDrone.Common.Http.JsonRpcRequestBuilder requestBuilder, System.String method, System.Object[] arguments) [0x0000e] in C:\projects\radarr-usby1\src\NzbDrone.Core\Download\Clients\Deluge\DelugeProxy.cs:213 
   --- End of inner exception stack trace ---
  at NzbDrone.Core.Download.Clients.Deluge.DelugeProxy.ExecuteRequest[TResult] (NzbDrone.Common.Http.JsonRpcRequestBuilder requestBuilder, System.String method, System.Object[] arguments) [0x0006f] in C:\projects\radarr-usby1\src\NzbDrone.Core\Download\Clients\Deluge\DelugeProxy.cs:234 
  at NzbDrone.Core.Download.Clients.Deluge.DelugeProxy.ProcessRequest[TResult] (NzbDrone.Core.Download.Clients.Deluge.DelugeSettings settings, System.String method, System.Object[] arguments) [0x00008] in C:\projects\radarr-usby1\src\NzbDrone.Core\Download\Clients\Deluge\DelugeProxy.cs:180 
  at NzbDrone.Core.Download.Clients.Deluge.DelugeProxy.SetLabel (System.String hash, System.String label, NzbDrone.Core.Download.Clients.Deluge.DelugeSettings settings) [0x00000] in C:\projects\radarr-usby1\src\NzbDrone.Core\Download\Clients\Deluge\DelugeProxy.cs:158 
  at NzbDrone.Core.Download.Clients.Deluge.Deluge.AddFromTorrentFile (NzbDrone.Core.Parser.Model.RemoteMovie remoteMovie, System.String hash, System.String filename, System.Byte[] fileContent) [0x00027] in C:\projects\radarr-usby1\src\NzbDrone.Core\Download\Clients\Deluge\Deluge.cs:54 
  at NzbDrone.Core.Download.TorrentClientBase`1[TSettings].DownloadFromWebUrl (NzbDrone.Core.Parser.Model.RemoteMovie remoteEpisode, System.String torrentUrl) [0x001b8] in C:\projects\radarr-usby1\src\NzbDrone.Core\Download\TorrentClientBase.cs:271 
  at NzbDrone.Core.Download.TorrentClientBase`1[TSettings].Download (NzbDrone.Core.Parser.Model.RemoteMovie remoteMovie) [0x00148] in C:\projects\radarr-usby1\src\NzbDrone.Core\Download\TorrentClientBase.cs:125 
  at NzbDrone.Core.Download.DownloadService.DownloadReport (NzbDrone.Core.Parser.Model.RemoteMovie remoteMovie, System.Boolean foceDownload) [0x000a3] in C:\projects\radarr-usby1\src\NzbDrone.Core\Download\DownloadService.cs:119 
  at NzbDrone.Api.Indexers.ReleaseModule.DownloadRelease (NzbDrone.Api.Indexers.ReleaseResource release) [0x00041] in C:\projects\radarr-usby1\src\NzbDrone.Api\Indexers\ReleaseModule.cs:73 
  at NzbDrone.Api.Indexers.ReleaseModule.<.ctor>b__8_0 (System.Object x) [0x00000] in C:\projects\radarr-usby1\src\NzbDrone.Api\Indexers\ReleaseModule.cs:47 
  at (wrapper dynamic-method) System.Object.CallSite.Target(System.Runtime.CompilerServices.Closure,System.Runtime.CompilerServices.CallSite,System.Func`2<object, object>,object)
  at Nancy.Routing.Route+<>c__DisplayClass4.<Wrap>b__3 (System.Object parameters, System.Threading.CancellationToken context) [0x00049] in <63c44d6f8d7e4f339a987bf9eb94d931>:0 
Close

**Comment by galli-leo** [comment]
Are you using a reverse proxy?

**Comment by rafisher79** [comment]
Nope - no reverse proxy here

**Comment by galli-leo** [comment]
Anything special about your config? (Deluge works fine for me, timeout usually indicates a failure to connect to deluge). You could also try downgrading mono to something like 5.2.

**Comment by rafisher79** [comment]
Nothing special at all.  It's funny though - in the download client setup it tests fine. Its just when its trying to import an actual torrent.

**Comment by galli-leo** [comment]
Hmm it's failing on setting the label in deluge. Have you messed with labels before? You might need to enable a plugin for it to work.

**Comment by wlsnchrs** [comment]
I was having a similar error with labels / categories not being applied even though the label plugin was enabled in Deluge. I recently updated Deluge and fiddled with some permissions before I started having this problem, so I can't speak to the root cause.

However, disabling and re-enabling the label plugin as well as re-adding the move after completed directories (these were reset for some reason) seems to have resolved the issue.

I am testing testing with a new torrent now to confirm this resolves the issue.
